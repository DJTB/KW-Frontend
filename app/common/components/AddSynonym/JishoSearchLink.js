import React from 'react';
import PropTypes from 'prop-types';

import { createJishoUrl } from 'common/api';
import IconLink from 'common/components/IconLink';

JishoSearchLink.propTypes = {
  keyword: PropTypes.string.isRequired,
  visuallyHidden: PropTypes.bool,
};

JishoSearchLink.defaultProps = {
  visuallyHidden: false,
};

function JishoSearchLink({ keyword, visuallyHidden, ...props }) {
  return (
    <IconLink
      href={createJishoUrl(keyword)}
      name="WORD_SEARCH"
      title={`Search Jisho for ${keyword}`}
      size="1.8rem"
      tabIndex={visuallyHidden ? -1 : 0}
      external
      plainLink
      visuallyHidden={visuallyHidden}
      {...props}
    />
  );
}

export default JishoSearchLink;
