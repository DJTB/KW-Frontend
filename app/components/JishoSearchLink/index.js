import React from 'react';
import PropTypes from 'prop-types';

import { createJishoUrl } from 'shared/urls';
import IconLink from 'components/IconLink';

JishoSearchLink.propTypes = {
  keyword: PropTypes.string.isRequired,
};

function JishoSearchLink({ keyword, ...props }) {
  return (
    <IconLink
      href={createJishoUrl(keyword)}
      name="WORD_SEARCH"
      title={`Search Jisho for ${keyword}`}
      size="1.8rem"
      tabIndex="-1"
      external
      plainLink
      {...props}
    />
  );
}

export default JishoSearchLink;
