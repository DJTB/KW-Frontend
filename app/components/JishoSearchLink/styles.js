import React from 'react';
import styled, { css } from 'styled-components';
import { timingFunctions } from 'polished';
import { ghost } from 'shared/styles/utils';
import A from 'base/A';

const style = css`
  display: block;
  margin: .2em .4em;
  transform: scale(.95);
  transition: transform .3s ${timingFunctions('easeInQuint')};
  &:hover {
    transform: scale(1);
    transition: transform .3s ${timingFunctions('easeOutQuint')};
  }
  ${({ visuallyHidden }) => visuallyHidden && ghost}
`;

export const StyledAnchor = styled(({
  visuallyHidden,
  ...props
}) => <A {...props} />)`${style}`;
