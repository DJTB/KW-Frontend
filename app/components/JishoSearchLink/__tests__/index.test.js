import "jest-styled-components";
import React from "react";
import { render } from "enzyme";

import JishoSearchLink from "../index";

describe("<JishoSearchLink />", () => {
  it("should match base snapshot", () => {
    const renderedComponent = render(<JishoSearchLink keyword="かな" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
