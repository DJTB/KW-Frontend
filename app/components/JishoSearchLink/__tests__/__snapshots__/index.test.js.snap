// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JishoSearchLink /> should match base snapshot 1`] = `
.c2 {
  display: block;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  position: relative;
  width: 1.8rem;
  height: 1.8rem;
  background-repeat: no-repeat;
  background-color: rgba(0,0,0,0);
  color: currentColor;
}

.c3 {
  display: block;
  pointer-events: none;
  -webkit-transform-origin: 50% 50% 0px;
  -ms-transform-origin: 50% 50% 0px;
  transform-origin: 50% 50% 0px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  fill: currentColor;
}

.c1 {
  text-decoration: none;
  color: inherit;
  cursor: pointer;
}

.c0 {
  -webkit-transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955),-webkit-transform 100ms linear;
  -webkit-transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955),transform 100ms linear;
  transition: all 250ms cubic-bezier(0.455,0.030,0.515,0.955),transform 100ms linear;
  cursor: pointer;
  opacity: .7;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c0:focus,
.c0:hover {
  opacity: 1;
  outline: none;
}

.c0:active {
  opacity: 1;
  -webkit-transform: scale(.9);
  -ms-transform: scale(.9);
  transform: scale(.9);
}

<a
  class="c0 c1"
  data-ignore-hotkeys="true"
  href="//jisho.org/search/かな"
  rel="external noopener noreferrer"
  tabindex="0"
  target="_blank"
  title="Search Jisho for かな"
>
  <span
    class="c2"
    color="currentColor"
  >
    <svg
      class="c3"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      title="WORD_SEARCH"
      viewBox="0 0 100 100"
      width="100%"
    >
      <path
        d="M80.53 25.85a2.67 2.67 0 0 0-.49-.05h-11V13.15a2 2 0 0 0-2.11-2L26 13.49a8.15 8.15 0 0 0-8.08 8.12v55.87A11.39 11.39 0 0 0 29.3 88.85h41.4a11.39 11.39 0 0 0 11.37-11.37V27.79a2 2 0 0 0-1.54-1.94zm-54.46-8.36h.11l38.91-2.22v10.52h-39a4.15 4.15 0 0 1 0-8.3zm52 60a7.38 7.38 0 0 1-7.37 7.37H29.3a7.38 7.38 0 0 1-7.37-7.37V28.66a8.09 8.09 0 0 0 4.14 1.13h52zM56 60.64a12.57 12.57 0 1 0-2.83 2.83l10.2 10.2a2 2 0 1 0 2.83-2.83zm-4.12-1.29a8.54 8.54 0 1 1 2.5-6 8.49 8.49 0 0 1-2.54 6z"
      />
    </svg>
  </span>
</a>
`;
